libpod/container_internal_linux.go

```diff
@@ -2225,6 +2249,92 @@ func (c *Container) copyOwnerAndPerms(source, dest string) error {
 	return nil
 }
 
+// harmlessHostPath validates if a host path is harmless and can be modified
+func (c *Container) harmlessHostPath(path string) bool {
+	excludePaths := map[string]bool{
+		"/":           true,
+		"/bin":        true,
+		"/boot":       true,
+		"/dev":        true,
+		"/etc":        true,
+		"/etc/passwd": true,
+		"/etc/pki":    true,
+		"/etc/shadow": true,
+		"/home":       true,
+		"/lib":        true,
+		"/lib64":      true,
+		"/media":      true,
+		"/opt":        true,
+		"/proc":       true,
+		"/root":       true,
+		"/run":        true,
+		"/sbin":       true,
+		"/srv":        true,
+		"/sys":        true,
+		"/tmp":        true,
+		"/usr":        true,
+		"/var":        true,
+		"/var/lib":    true,
+		"/var/log":    true,
+	}
+
+	if home := os.Getenv("HOME"); home != "" {
+		excludePaths[home] = true
+	}
+
+	if sudoUser := os.Getenv("SUDO_USER"); sudoUser != "" {
+		if usr, err := user.Lookup(sudoUser); err == nil {
+			excludePaths[usr.HomeDir] = true
+		}
+	}
+
+	if path != "/" {
+		path = strings.TrimSuffix(path, "/")
+	}
+
+	if excludePaths[path] {
+		return false
+	}
+
+	return true
+}
+
+// chownSourceVolume changes the ownership of a source volume.
+func (c *Container) chownSourceVolume(path string, UID, GID int) error {
+	// Validate if shource volume can be chowned
+	if !c.harmlessHostPath(path) {
+		return errors.Errorf("chowning source volume %q is not allowed", path)
+	}
+
+	// Get source volume info
+	fi, err := os.Lstat(path)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get path info when chowning source volume")
+	}
+
+	// Get current ownership
+	currentUID := int(fi.Sys().(*syscall.Stat_t).Uid)
+	currentGID := int(fi.Sys().(*syscall.Stat_t).Gid)
+
+	// Chown source volume
```

Comment:
Isn't the exact same code in Buildah?  Can you break it out so that Podman and Buildah can share the same Chowning code.

Reply:
Yes, this is the same code but sadly I did not create the method with capital letter so it can not be exported. I do not have any problem in sending another PR to Buildah to change that, so let me know the best way to proceed with this. thanks

Reply:
Let's do that.